### PR TITLE
Apply Clippy

### PR DESCRIPTION
**Pull request by sanxiyn**

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -1008,7 +1008,7 @@ fn rewrite_guard(context: &RewriteContext,
                  // the arm (excludes offset).
                  pattern_width: usize)
                  -> Option<String> {
-    if let &Some(ref guard) = guard {
+    if let Some(ref guard) = *guard {
```
Comment (nrc) [inline src/expr.rs]: I think this is somewhat dubious advice for `if let`, I guess at least it is consistent with match, but I feel the original is clearer

File: src/missed_spans.rs [hunk 2]
```diff
@@ -94,9 +94,10 @@ impl<'a> FmtVisitor<'a> {
         fn replace_chars(string: &str) -> String {
             string.chars()
                   .map(|ch| {
-                      match ch.is_whitespace() {
-                          true => ch,
-                          false => 'X',
+                      if ch.is_whitespace() {
+                          ch
+                      } else {
+                          'X'
```
Comment (nrc) [inline src/missed_spans.rs]: also dubious - the `match` is shorter, so why is `if else` preferred?

**Comment by nrc** [comment]
Thanks!

File: src/missed_spans.rs (hunk 2 above)
Comment (sanxiyn) [inline src/missed_spans.rs]: Well, in a sense, `match` is shorter than `if else` here because rustfmt formats so...

**Pull request merged**